# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/daniyalhaider236/flyrank-ml/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# 1. My Lane and Why

I chose **Lane 2: Refresh / Content Opportunity Scoring** because it focuses on helping content teams decide which pages should be reviewed first for content updates.

Instead of reviewing every page manually, I want to investigate whether observable content and search metrics can be used to prioritise pages that are more likely to need a refresh. This can help make content review more efficient while keeping the reasoning behind each recommendation transparent.

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# 2. The Question

### Research Question

Can observable page metrics such as content age, impressions, CTR, average position, and update history help identify pages that should be prioritised for content refresh?

### Decision

Which pages should be reviewed first for content refresh?

### Action

Editors or SEO teams review and update the highest-priority pages before spending time on lower-priority pages.

### Cost of a Wrong Recommendation

If a page is incorrectly recommended for refresh, editorial time may be wasted on content that does not need updating.

If an important declining page is missed, valuable search traffic and user engagement may continue to decrease.

### Why Data or Machine Learning Helps

A large website can contain hundreds or thousands of pages. Looking at several performance signals together is difficult manually. A transparent machine learning model can rank pages based on multiple observable features and provide consistent recommendations while still allowing humans to make the final decision.

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:


df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))

print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:")
print(df["ctr"].mean())

print("\nAverage Content Age:")
print(df["content_age_days"].mean())


Number of pages: 30000

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR:
0.5107333333333334

Average Content Age:
256.1678


# 3. Quick Look at the Data

The starter dataset contains **30000 pages**.

The distribution of **trend_direction** shows that the dataset contains both improving and declining pages, making it suitable for studying content refresh prioritisation.

The average CTR is **0.510733333333334**, while the average content age is **256.1678 days**.

These statistics suggest there is meaningful variation in page performance and freshness, making this a useful problem to investigate over the next several weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# 4. Careful Words

This project will analyse relationships within the available dataset and build models that prioritise pages for review.

The results will describe patterns observed in this data only.

The project will not claim to have discovered Google's ranking algorithm or prove that any feature directly causes better rankings or more traffic.

Any recommendations produced by the model should be considered decision-support rather than automatic decisions.

In [17]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

# 5. Self-Check

- ✔ I selected one of the predefined lanes.
- ✔ I defined a clear research question.
- ✔ I explained the decision the project supports.
- ✔ I described the action someone can take.
- ✔ I explained the cost of an incorrect recommendation.
- ✔ I supported my choice using real numbers from the starter dataset.
- ✔ I avoided making causal claims or overstating the model's capabilities.